docs/operation/upgrade.rst

```diff
@@ -0,0 +1,94 @@
+Upgrade Guide
+=============
+This document describes a complete and reliable upgrade path for **MetalK8s**
+and all the components that make up the stack.
+
+.. important::
+
+    - Upgrading MetalK8s is a manual process and as a result,
+      the following outlined steps must be met to the later in other to achieve
+      a successful upgrade.
+    - The output of the below-provided commands solely depends on the particular
+      MetalK8s setup you have in place.
+
+.. warning::
+    For any cluster upgrade from MetalK8s v2.0 to v2.1 and on-ward, **Kubelet**
+    must be upgraded before making any upgrades to the salt-master. Once this
+    is done, an upgrade is guaranteed to proceed in a seamless manner.
+
+Supported Versions
+******************
+
+This guide currently supports MetalK8s upgrades for the following versions.
+Please refer to the release notes for more information.
+
+    - Upgrade 2.0.0 to 2.1.0
+    - Upgrade 2.1.0 to 2.2.0
+
+
+Upgrade Steps
+*************
+This section describes the logical steps for Upgrading MetalK8s.
+
+#. Connect to the :term:`Bootstrap node`.
+
+#. Download the new `.ISO` from `Sources <http://scality.com>`_.
+
+#. Edit the `bootstrap.yaml` file located in `/etc/metalk8s/bootstrap.yaml`
+   then add the new MetalK8s ISO location.
+
+   .. code::
+
+      products:
+        metalk8s:
+        - /srv/scality/metalk8s-2.0.0-dev
+        - /root/metalk8s-<new_version>.iso     <=== New line to add
+
+
+#. Mount the new MetalK8s ISO.
+
+    ::
+
+        salt-call state.sls metalk8s.products.mounted saltenv=metalk8s-<current_version>
+
+            Summary for local
+            ------------
+            Succeeded: 5 (changed=1)
+            Failed:    0
+            ------------
+            Total states run:     5
+
+#. Advertise the new version and make it available to the entire cluster.
+
+    ::
+
+        salt-call --local state.sls metalk8s.products saltenv=metalk8s-<new_version> pillar="{'metalk8s': {'endpoints': $(salt-call --out txt pillar.get metalk8s:endpoints | cut -c 8-)}}"
+
+#. From the :term:`Bootstrap node`, connect to the :term:`Salt Master`.
+
+    ::
+
+        kubectl -n kube-system exec -it salt-master-bootstrap bash
+
+    .. warning::
+        The next few commands must be executed only on the :term:`Salt Master`.
+        Before proceeding, make sure that this is verified.
+
+#. From the :term:`Salt Master`, launch the rolling upgrade procedure.
+
+    ::
+
+        salt-run state.orchestrate metalk8s.orchestrate.upgrade saltenv=metalk8s-<new_version> pillar="{'orchestrate': {'dest_version': '<new_version>'}}"
+
+    .. note::
+        By default, the orchestrate mechanism will upgrade pods and cluster
+        components on a one by one basis.
```

Comment:
We should also say that it's take time and we have no output until the upgrade totaly finished